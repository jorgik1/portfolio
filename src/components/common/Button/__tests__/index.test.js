import React from 'react';
import { render, screen } from '@testing-library/react';
import { Button } from '..'; // Assuming the Button component is in the parent directory

describe('Button Component', () => {
  it('renders default button correctly and matches snapshot', () => {
    render(<Button>Click Me</Button>);
    const buttonElement = screen.getByText(/Click Me/i);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });

  it('renders secondary button correctly and matches snapshot', () => {
    render(<Button secondary>Secondary Action</Button>);
    const buttonElement = screen.getByText(/Secondary Action/i);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });
});
