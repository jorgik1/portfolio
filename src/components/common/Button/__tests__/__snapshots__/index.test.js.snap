// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders default button correctly and matches snapshot 1`] = `
<button
  class="sc-bdVaJa hYVUdD"
>
  Click Me
</button>
`;

exports[`Button Component renders secondary button correctly and matches snapshot 1`] = `
<button
  class="sc-bdVaJa hMlUGK"
>
  Secondary Action
</button>
`;
